R/tm_a_pca.R

```diff
@@ -246,18 +246,19 @@ ui_a_pca <- function(id, ...) {
 srv_a_pca <- function(id, data, reporter, filter_panel_api, dat, plot_height, plot_width, ggplot2_args) {
   with_reporter <- !missing(reporter) && inherits(reporter, "Reporter")
   with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelAPI")
-  checkmate::assert_class(data, "tdata")
+  checkmate::assert_class(data, "reactive")
+  checkmate::assert_class(isolate(data()), "teal_data")
   moduleServer(id, function(input, output, session) {
     response <- dat
 
     for (i in seq_along(response)) {
       response[[i]]$select$multiple <- FALSE
       response[[i]]$select$always_selected <- NULL
       response[[i]]$select$selected <- NULL
-      response[[i]]$select$choices <- var_labels(data[[response[[i]]$dataname]]())
+      response[[i]]$select$choices <- var_labels(data()[[response[[i]]$dataname]])
```

Comment:
This is outside of a reactive contect, we should isolate this (I know there was reactive evaluation before, but this is wrong).
```suggestion
      response[[i]]$select$choices <- var_labels(isolate(data())[[response[[i]]$dataname]])
```